app/models/foreman_rh_cloud/ping.rb

```diff
@@ -0,0 +1,85 @@
+module ForemanRhCloud
+  class Ping
+    OK_RETURN_CODE = 'ok'.freeze
+    FAIL_RETURN_CODE = 'FAIL'.freeze
+
+    SERVICE_URLS = {
+      :advisor => "https://localhost:24443/api/insights/v1/status/live/",
+      :vulnerability => "https://localhost:24443/api/vulnerability/v1/apistatus"
+    }
+
+    class << self
+      def services
+        SERVICE_URLS.keys
+      end
+
+      def status
+        {
+          version: ForemanRhCloud::VERSION,
+          timeUTC: Time.now.getutc,
+        }
+      end
+
+      def exception_watch(result, &blk)
+        ::Katello::Ping.exception_watch(result, &blk)
+      end
+
+      def ping
+        ping_services
+      end
+
+      def ping!
+        result = ping_services
+
+        if result[:status] != OK_RETURN_CODE
+          failed_names = result[:services].select do |_name, details|
+            details[:status] != OK_RETURN_CODE
+          end
+          fail "The following services have not been started or are reporting errors: #{failed_names.keys.join(', ')}"
+        end
+
+        result
+      end
+
+      def ping_services
+        result = {}
+        services.each do |service|
+          result[service] = {}
+          ping_service(service, result[service])
+        end
+
+        # set overall status result code
+        result = {:services => result}
+        result[:status] = result[:services].each_value.any? { |v| v[:status] == FAIL_RETURN_CODE } ? FAIL_RETURN_CODE : OK_RETURN_CODE
+        result
+      end
+
+      def ping_url(url)
+        ca_file = Setting[:ssl_ca_file]
+        request_id = ::Logging.mdc['request']
+
+        options = {}
+        options[:ssl_ca_file] = ca_file unless ca_file.nil?
```

Comment:
You can DRY it by reusing https://github.com/theforeman/foreman_rh_cloud/blob/develop/app/services/foreman_rh_cloud/cert_auth.rb concern in the file.